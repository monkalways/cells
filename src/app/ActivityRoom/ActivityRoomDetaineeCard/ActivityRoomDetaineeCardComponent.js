import React from 'react';
import PropTypes from 'prop-types';
import { Card, withStyles } from '@material-ui/core';

import {
  CellDetaineeCardHeader,
  CellDetaineeCardContent,
} from '../../common/CellDetaineeCard';
import CardFooter from './CardFooter';

const propTypes = {
  classes: PropTypes.shape({}).isRequired,
  detainee: PropTypes.shape({
    firstName: PropTypes.string.isRequired,
    lastName: PropTypes.string.isRequired,
  }).isRequired,
  usage: PropTypes.string.isRequired,
  isAuthenticated: PropTypes.bool.isRequired,
  isCheckingIn: PropTypes.bool.isRequired,
  onCheckIn: PropTypes.func.isRequired,
};

const ActivityRoomDetaineeCardComponent = ({
  classes,
  detainee,
  usage,
  isAuthenticated,
  isCheckingIn,
  onCheckIn,
}) => (
  <Card className={classes.card}>
    <CellDetaineeCardHeader cellDetainee={detainee} />
    <CellDetaineeCardContent
      cellDetainee={detainee}
      first="activity-rooms"
      second={usage}
      isAuthenticated={isAuthenticated}
    />
    <CardFooter
      usage={usage}
      detainee={detainee}
      isAuthenticated={isAuthenticated}
      isCheckingIn={isCheckingIn}
      onCheckIn={onCheckIn}
    />
  </Card>
);

ActivityRoomDetaineeCardComponent.propTypes = propTypes;

export default withStyles((theme) => ({
  card: {
    maxWidth: theme.spacing.unit * 50,
    backgroundColor: theme.palette.background.paper,
  },
}))(ActivityRoomDetaineeCardComponent);
