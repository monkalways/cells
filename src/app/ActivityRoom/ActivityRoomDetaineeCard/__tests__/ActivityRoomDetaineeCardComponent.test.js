import React from 'react';
import { shallow } from 'enzyme';

import { ActivityRoomDetaineeCardComponent } from '../ActivityRoomDetaineeCardComponent';

describe('ActivityRoomDetaineeCardComponent', () => {
  it('should render', () => {
    const classes = { card: {} };
    const detainee = {};
    const wrapper = shallow(<ActivityRoomDetaineeCardComponent
      classes={classes}
      detainee={detainee}
      usage="phone"
      isAuthenticated={false}
      isCheckingIn={false}
      onCheckIn={jest.fn()}
    />);

    expect(wrapper).toMatchSnapshot();
  });
});
