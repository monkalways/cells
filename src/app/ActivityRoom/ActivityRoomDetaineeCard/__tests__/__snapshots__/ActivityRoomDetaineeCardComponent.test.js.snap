// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActivityRoomDetaineeCardComponent should render 1`] = `
<WithStyles(Card)
  className={Object {}}
>
  <WithStyles(CellDetaineeCardHeaderComponent)
    cellDetainee={Object {}}
  />
  <WithStyles(withRouter(CellDetaineeCardContentComponent))
    cellDetainee={Object {}}
    first="activity-rooms"
    isAuthenticated={false}
    second="phone"
  />
  <WithStyles(withRouter(CardFooterComponent))
    detainee={Object {}}
    isAuthenticated={false}
    isCheckingIn={false}
    onCheckIn={[MockFunction]}
    usage="phone"
  />
</WithStyles(Card)>
`;
